src/cljs/rems/new_application.cljs

```diff
@@ -2,49 +2,42 @@
   (:require [re-frame.core :as rf]
             [rems.application :refer [navigate-to]]
             [rems.spinner :as spinner]
-            [rems.status-modal :refer [status-modal]]
+            [rems.status-modal :as status-modal]
             [rems.text :refer [text text-format]]
-            [rems.util :refer [fetch post!]]))
-
-(rf/reg-event-db
- ::set-status
- (fn [db [_ {:keys [status description error] :as state}]]
-   (assoc db ::status state)))
-
-(rf/reg-sub ::status (fn [db _] (::status db)))
+            [rems.util :refer [dispatch! fetch post!]]))
 
 (defn remove-catalogue-items-from-cart! [catalogue-item-ids]
   (doseq [i catalogue-item-ids]
     (rf/dispatch [:rems.cart/remove-item i])))
 
-(defn- save-application [description catalogue-item-ids on-success]
+(defn- save-application [[description catalogue-item-ids on-success]]
   (post! "/api/applications/save"
          {:handler (fn [response]
                      (if (:success response)
                        (on-success response)
-                       (rf/dispatch [::set-status {:status :failed
-                                                   :description description}])))
-          :error-handler (fn [error]
-                           (rf/dispatch [::set-status {:status :failed
-                                                       :description description
-                                                       :error error}]))
+                       (status-modal/set-error! {:result response})))
+          :error-handler #(status-modal/set-error! {:result {:error %}})
```

Comment:
it's a bit surprising that set-error! takes the whole state (instead of just the `%` here), but I guess it's ok

Reply:
In this case we know our API call returns the suitable structure with `:errors` keys. It could be explicitly destructured but I opted for convention here. The `:error-handler` mostly returns HTTP errors ant that's where the structure must be explicitly transformed.